content/spaces/locking.textile

```diff
@@ -0,0 +1,301 @@
+---
+title: Component locking
+meta_description: "Component locking enables members to lock UI components before editing them to reduce the chances of conflicting changes being made."
+languages:
+  - javascript
+---
+
+The component locking feature enables members to optimistically lock stateful UI components before editing them. This reduces the chances of conflicting changes being made to the same component by different members. A component could be a cell in a spreadsheet that a member is updating, or an input field on a form they're filling in.
+
+Once a lock has been acquired by a member, the component that it relates to can be updated in the UI to visually indicate to other members that it is locked and and which member has the lock. The component can then be updated once the editing member has released the lock to indicate that it is now free.
+
+Each lock is identified by a unique string ID, and only a single member may hold a lock with a given string at any one time. A lock will exist in one of three "states":#states and may only transition between states in specific circumstances.
+
+<aside data-type='important'>
+<p>Optimistic locking means that there is a chance that two members may begin editing the same UI component before it is confirmed which member holds the lock. On average, the time taken to reconcile which member holds a lock is in the hundreds of milliseconds. Your application needs to handle the member that successfully obtained the lock, as well as the member that had their request invalidated.</p>
+</aside>
+
+h2(#states). Lock states
+
+Component locking is handled entirely client-side. Members will optimistically begin to edit a component as soon as they call "acquire()":#acquire on the lock identifier related to it. A short period, on average in the hundreds of milliseconds, will follow before it is confirmed whether the lock request was successful or rejected.
```

Comment:
> Members will optimistically begin to edit a component as soon as they call "acquire()":#acquire on the lock identifier related to it

This isn't necessarily true, a developer could wait for the lock to transition to LOCKED before allowing edits, and for example show a spinner until that state is reached. This is still optimistic locking, because the LOCKED state may transition to UNLOCKED shortly after if some other member also acquired the lock just before but that presence event hasn't propagated yet.

Reply:
Ah that's interesting. I've reworded to account for both scenarios in [0f7e2bd](https://github.com/ably/docs/pull/1967/commits/0f7e2bdebb72e8a3f291c97deca282f36af69782), but if you think we should suggest one approach over the other then let me know.

Reply:
LGTM 👍